File: Backend/Salud360/src/main/java/pe/edu/pucp/salud360/membresia/models/Membresia.java
package pe.edu.pucp.salud360.membresia.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pe.edu.pucp.salud360.comunidad.models.Comunidad;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "membresia")
@Inheritance(strategy = InheritanceType.JOINED)
public class Membresia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idMembresia;

    @Column(name = "nombre", unique = false, nullable = false, updatable = true)
    private String nombre;

    @Column(name = "descripcion", unique = false, nullable = false, updatable = true)
    private String descripcion;

    @Column(name = "tipo", unique = false, nullable = false, updatable = true)
    private String tipo;

    @Column(name = "conTope", unique = false, nullable = false, updatable = true)
    private Boolean conTope;

    @Column(name = "precio", unique = false, nullable = false, updatable = true)
    private Double precio;

    @Column(name = "cantUsuarios", unique = false, nullable = false, updatable = true)
    private Integer cantUsuarios;

    @Column(name = "icono", unique = false, nullable = false, updatable = true)
    private String icono;

    @Column(name = "activo", unique = false, nullable = false, updatable = true)
    private Boolean activo;

    @Column(name = "fechaCreacion", unique = false, nullable = false, updatable = true)
    private LocalDateTime fechaCreacion;

    @Column(name = "fechaDesactivacion", unique = false, nullable = false, updatable = true)
    private LocalDateTime fechaDesactivacion;

    @ManyToOne
    @JoinColumn(name = "idComunidad")
    private Comunidad comunidad;
}

File: Frontend/Salud360/src/schemas/membresia.ts
import { DateTime } from "luxon";
import { z } from "zod";

export const membresiaSchema = z.object({
  idMembresia: z.number(),
  nombre: z.string(),
  descripcion: z.string(),
  tipo: z.string(),
  conTope: z.boolean(),
  precio: z.number(),
  cantUsuariosinteger: z.number(),
  maxReservasinteger: z.number(),
  icono: z.string(),
  activo: z.boolean(),
  fechaCreacion: z.string().transform((v: string) => DateTime.fromISO(v)),
  fechaDesactivacion: z.string().transform((v: string) => DateTime.fromISO(v)),
});

export type membresiaType = z.infer<typeof membresiaSchema>;
